datafusion/core/tests/sql/decimal.rs

```diff
@@ -0,0 +1,428 @@
+        "| 0.000000000004 | 12  |",
+        "| 0.000000000004 | 14  |",
+        "| 0.000000000004 | 8   |",
+        "| 0.000000000005 | 9   |",
+        "| 0.000000000005 | 4   |",
+        "| 0.000000000005 | 8   |",
+        "| 0.000000000005 | 100 |",
+        "| 0.000000000005 | 1   |",
+        "+----------------+-----+",
+    ];
+    assert_batches_eq!(expected, &actual);
+    // logic operation: lt
+    let sql = "select * from decimal_simple where c1 < 0.00002";
+    let actual = execute_to_batches(&ctx, sql).await;
+    let expected = vec![
+        "+----------+----------------+----+------+-----------+",
+        "| c1       | c2             | c3 | c4   | c5        |",
+        "+----------+----------------+----+------+-----------+",
+        "| 0.000010 | 0.000000000001 | 1  | true | 0.0000140 |",
+        "+----------+----------------+----+------+-----------+",
+    ];
+    assert_batches_eq!(expected, &actual);
+
+    // logic operation: lteq
+    let sql = "select * from decimal_simple where c1 <= 0.00002";
+    let actual = execute_to_batches(&ctx, sql).await;
+    let expected = vec![
+        "+----------+----------------+----+-------+-----------+",
+        "| c1       | c2             | c3 | c4    | c5        |",
+        "+----------+----------------+----+-------+-----------+",
+        "| 0.000010 | 0.000000000001 | 1  | true  | 0.0000140 |",
+        "| 0.000020 | 0.000000000002 | 2  | true  | 0.0000250 |",
+        "| 0.000020 | 0.000000000002 | 3  | false | 0.0000190 |",
+        "+----------+----------------+----+-------+-----------+",
+    ];
+    assert_batches_eq!(expected, &actual);
+
+    // logic operation: gt
+    let sql = "select * from decimal_simple where c1 > 0.00002";
+    let actual = execute_to_batches(&ctx, sql).await;
+    let expected = vec![
+        "+----------+----------------+-----+-------+-----------+",
+        "| c1       | c2             | c3  | c4    | c5        |",
+        "+----------+----------------+-----+-------+-----------+",
+        "| 0.000030 | 0.000000000003 | 4   | true  | 0.0000320 |",
+        "| 0.000030 | 0.000000000003 | 5   | false | 0.0000350 |",
+        "| 0.000030 | 0.000000000003 | 5   | true  | 0.0000110 |",
+        "| 0.000040 | 0.000000000004 | 5   | true  | 0.0000440 |",
+        "| 0.000040 | 0.000000000004 | 12  | false | 0.0000400 |",
+        "| 0.000040 | 0.000000000004 | 14  | true  | 0.0000400 |",
+        "| 0.000040 | 0.000000000004 | 8   | false | 0.0000440 |",
+        "| 0.000050 | 0.000000000005 | 9   | true  | 0.0000520 |",
+        "| 0.000050 | 0.000000000005 | 4   | true  | 0.0000780 |",
+        "| 0.000050 | 0.000000000005 | 8   | false | 0.0000330 |",
+        "| 0.000050 | 0.000000000005 | 100 | true  | 0.0000680 |",
+        "| 0.000050 | 0.000000000005 | 1   | false | 0.0001000 |",
+        "+----------+----------------+-----+-------+-----------+",
+    ];
+    assert_batches_eq!(expected, &actual);
+
+    // logic operation: gteq
```

Comment:
Is it worth checking somewhere what happens with different decimal scales?

Something like
```sql 
select * from decimal_simple where c1 >= CAST(0.00002 as Decimal(10,8))
```

I am not sure if the coercion to an appropriate decimal type is covered elsewhere

Reply:
good point.
c1 is decimal(10,6), and the cast(xxx as decimal(10,8)) is decimal(10,8), the result type should be the `decimal(12,8)`.
I will add this test case.

Reply:
I find the bug in the precision and scale for logic comparison operation.
```
❯ explain select * from food where a >= CAST(0.00002 as Decimal(10,8));
+---------------+----------------------------------------------------------------------------------------------------------+
| plan_type     | plan                                                                                                     |
+---------------+----------------------------------------------------------------------------------------------------------+
| logical_plan  | Projection: #food.a, #food.b, #food.c                                                                    |
|               |   Filter: #food.a >= Decimal128(Some(2000),10,8)                                                         |
|               |     TableScan: food projection=Some([0, 1, 2]), partial_filters=[#food.a >= Decimal128(Some(2000),10,8)] |
| physical_plan | ProjectionExec: expr=[a@0 as a, b@1 as b, c@2 as c]                                                      |
|               |   CoalesceBatchesExec: target_batch_size=4096                                                            |
|               |     FilterExec: CAST(a@0 AS Decimal(10, 8)) >= Some(2000),10,8                                           |
|               |       RepartitionExec: partitioning=RoundRobinBatch(16)                                                  |
|               |         CsvExec: files=[aggregate_simple.csv], has_header=false, limit=None, projection=[a, b, c]        |
|               |                                                                                                          |
+---------------+----------------------------------------------------------------------------------------------------------+
```
@alamb

Reply:
https://github.com/apache/arrow-datafusion/issues/2232

Reply:
> I find the bug in the precision and scale for logic comparison operation.

Testing for the win!